src/solvers/refinement/string_constraint_generator_valueof.cpp

```diff
@@ -437,12 +439,14 @@ exprt string_constraint_generatort::add_axioms_for_parse_int(
     not_exprt(or_exprt(starts_with_minus, starts_with_plus));
 
   /// TODO: we should throw an exception when this does not hold:
-  exprt correct=add_axioms_for_correct_number_format(str, radix);
+  std::size_t max_string_length=40;
```

Comment:
maybe we could use the value given by the `--max-string-length` option here, `max_string_length` is already a variable defined in this class, you can take the minimum value between this and 40 for instance.